pkg/local_object_storage/shard/get.go

```diff
@@ -102,40 +102,42 @@ func (s *Shard) fetchObjectData(addr oid.Address, skipMeta bool, cb storFetcher,
 		mRes meta.ExistsRes
 	)
 
-	var exists bool
 	if !skipMeta {
 		var mPrm meta.ExistsPrm
 		mPrm.SetAddress(addr)
-
 		mRes, mErr = s.metaBase.Exists(mPrm)
 		if mErr != nil && !s.info.Mode.NoMetabase() {
 			return nil, false, mErr
 		}
-		exists = mRes.Exists()
+
+		if !mRes.Exists() {
```

Comment:
To we need to take `s.info.Mode.NoMetabase()` into account here?

Reply:
https://github.com/TrueCloudLab/frostfs-node/blob/5303736acd240809c67847c2c898f97a3e419d05/pkg/local_object_storage/shard/get.go#L85

that line save us from that `if` branch execution without meta